### PR TITLE
Bump Go to 1.19

### LINKED ISSUE #6581: Use Go 1.20 (or 1.19)
1.20 was released on Feb 1st. This means that our current version (1.18) is no longer supported.

- [ ] Update all our CI jobs to use 1.20 (or at least 1.19 if 1.20 is not available)
- [ ] Update release jobs to use 1.20 (or at least 1.19 if 1.20 is not available)

/priority high

### PR DESCRIPTION
**Pull request by rm3l**
**What type of PR is this:**
/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**
Fixes #6581 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by rm3l** [comment]
/hold

Until we have checked with the release team that Go 1.19 is available.

**Review by feloy — APPROVED** [review]

**Comment by rm3l** [comment]
/hold cancel

Removing the hold, since [CWFCONF-3455](https://issues.redhat.com/browse/CWFCONF-3455) has been resolved and I've checked that Golang 1.19 was available internally.
Also rebased to test that the IBM Cloud image builds successfully (https://github.com/redhat-developer/odo/pull/6587).

**Review by feloy — APPROVED** [review]

**Pull request merged**
